compiler/fir/tree/src/org/jetbrains/kotlin/fir/BadNamedArgumentsTarget.kt

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+
+package org.jetbrains.kotlin.fir
```

Comment:
This place seems very vague to me. Also not sure why it belongs to `tree`. I think `...fir.resolve.diagnostics` in `resolve` module would be better.

Reply:
Put it in resolve will make the reference in FirDiagnosticsList unresolvable. We can use a string there instead if we really don't want to have this enum under tree.

Reply:
Oh, I saw that `checker-component-generator` has minimal dependencies. @darthorimar , doesn't it make sense to make that module depend on `resolve` module, since generated diagnostic list, as part of `checkers`, eventually use `resolve` (and placed literally after `resolve`)?

It still doesn't make sense to me to put this in `tree` because of that dependency issue.

Reply:
`BadNamedArgumentsTarget` should be also be accessible from IDE HIgh-Level API as a parameter of generated `KtDiagnostic`. As High-Level API should now nothing about specific frontend and should not depend on FIR modules, we should consider placing `BadNamedArgumentsTarget` into `:compiler:frontend.common` module

Reply:
Thanks! Moved to `:compiler:frontend.common`.